10-habits/progress/internal/server/getstatus.go

```diff
@@ -0,0 +1,27 @@
+package server
+
+import (
+	"context"
+	"log"
+
+	"learngo-pockets/habits/api"
+	"learngo-pockets/habits/internal/habit"
+)
+
+// GetHabitStatus is the endpoint that retrieves the status of a habit per week.
+func (s *Server) GetHabitStatus(ctx context.Context, request *api.GetHabitStatusRequest) (*api.GetHabitStatusResponse, error) {
+	log.Printf("GetHabitStatus request received: %s", request)
+	h, ticksCount, err := habit.GetHabitStatus(ctx, s.db, s.tickDB, habit.ID(request.HabitId))
```

Comment:
Maybe this should be in charge of telling the current time (rather than letting the DB layer call `time.Now()`

Reply:
Changed on the server side.